tests/test_iri.py

```diff
@@ -0,0 +1,148 @@
+    )
+
+
+@pytest.fixture()
+def mock_download(mocker):
+    """
+    Call download with mocked _download.
+
+    `forecast_type` is the type of forecast to
+    test, can be either 'prob' or 'dominant'.
+    """
+    download_mock = mocker.patch(
+        "aatoolbox.datasources.iri."
+        "iri_seasonal_forecast._IriForecast._download"
+    )
+
+    def _mock_download(forecast_type):
+        geo_bounding_box = GeoBoundingBox(north=6, south=3.2, east=-2, west=3)
+        if forecast_type == "prob":
+            iri_class = IriForecastProb(
+                iso3=ISO3, geo_bounding_box=geo_bounding_box
+            )
+        elif forecast_type == "dominant":
+            iri_class = IriForecastDominant(
+                iso3=ISO3, geo_bounding_box=geo_bounding_box
+            )
+        iri_class.download(iri_auth=None)
+        _, kwargs_download = download_mock.call_args
+        url = kwargs_download["url"]
+        filepath = kwargs_download["filepath"]
+        return url, filepath
+
+    return _mock_download
+
+
+def test_download_call_prob(mock_download):
+    """Test download for tercile probability forecast."""
+    url, filepath = mock_download("prob")
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+        ".NMME_Seasonal_Forecast/"
+        ".Precipitation_ELR/.prob/X/%283.0%29%28-2.0%29RANGEEDGES/"
+        "Y/%286.0%29%283.0%29RANGEEDGES/data.nc"
+    )
+
+    assert filepath == (
+        Path(FAKE_AA_DATA_DIR) / f"private/raw/{ISO3}/{_MODULE_BASENAME}/"
+        f"abc_iri_forecast_seasonal_precipitation_"
+        f"tercile_prob_Np6Sp3Em2Wp3.nc"
+    )
+
+
+def test_download_call_dominant(mock_download):
+    """Test download for dominant tercile forecast."""
+    url, filepath = mock_download("dominant")
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+        ".NMME_Seasonal_Forecast/"
+        ".Precipitation_ELR/.dominant/X/%283.0%29%28-2.0%29RANGEEDGES/"
+        "Y/%286.0%29%283.0%29RANGEEDGES/data.nc"
+    )
+
+    assert filepath == (
+        Path(FAKE_AA_DATA_DIR) / f"private/raw/{ISO3}/{_MODULE_BASENAME}/"
+        f"abc_iri_forecast_seasonal_"
+        f"precipitation_tercile_dominant_Np6Sp3Em2Wp3.nc"
+    )
+
+
+def test_process(mocker):
+    """Test process for IRI forecast."""
+    ds = xr.DataArray(
+        np.reshape(a=np.arange(16), newshape=(2, 2, 2, 2)),
+        coords=[
+            [1, 2],
+            [3, -2],
+            [97, 90],
+            [685.5, 686.5],
+        ],
+        dims=["L", "X", "Y", "F"],
+    ).to_dataset(name="prob")
+    ds["F"].attrs["calendar"] = "360"
+    ds["F"].attrs["units"] = "months since 1960-01-01"
+    geo_bounding_box = GeoBoundingBox(north=6, south=3.2, east=-2, west=3)
+    iri_class = IriForecastProb(iso3=ISO3, geo_bounding_box=geo_bounding_box)
+
+    # TODO: now created `load_raw` to be able to mock but would like
+    # to do it from xr.load_dataset directly
+    mocker.patch(
```

Comment:
Here I think you can just do  `mocker.patch.object(xr, "load_dataset", return_value=ds)`. I don't totally understand why; but I think your method might work if you didn't import `xarray` into this module.

Reply:
mehehe your suggestion (which is cool!) seems to return the same result as `mocker.patch("aatoolbox.datasources.iri.iri_seasonal_forecast.xr.load_dataset",return_value=ds)` , i.e. `ds` is not being processed.. 

Why do you think your suggestion should work and the other not?

Reply:
@turnerm should we leave it like it is now (with the separate `load_raw` func) or do you have a better idea still? Else will merge

Reply:
Ah yes `mocker.patch("aatoolbox.datasources.iri.iri_seasonal_forecast.xr.load_dataset",return_value=ds) ` might even be better. I gave that a try, and it is actually mocking out `xr.load_dataset` properly, it's just returning something different than `load_raw`, which is very weird!

For example, in the first assertion if you change it to `[3, 2]` then it works. Which I guess means that somehow the `_process` steps aren't running on it or something? Let me investigate a bit more...

Reply:
aah sorry yes that is exactly what I meant. It runs fine, but it doesn't seem to go through _process (especially clear when you look at the values of the `F` dimension, which should change substantially)